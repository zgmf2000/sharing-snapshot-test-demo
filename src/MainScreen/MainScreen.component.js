/**
 * Sample React Native MainScreen
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Fragment, useState, Component} from 'react';
import {SafeAreaView, StatusBar, StyleSheet, Text, View} from 'react-native';
import Button from '../Component/Button';
import CounterText from '../Component/CounterText';

export default class MainScreen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            count: 0,
        };
    }

    setCount = (newValue) => {
        this.setState({
            count: newValue,
        });
    };

    //TODO: Show the depedencies using snapshot
    render() {
        const {count} = this.state;
        return (
            <Fragment>
                <StatusBar barStyle="dark-content"/>
                <SafeAreaView>
                    <View style={{marginTop: 25}}>
                        <View style={{alignItems: 'center'}}>
                            <CounterText count={count}/>
                        </View>
                        <View
                            style={style.controllerViewStyle}
                        >
                            <Button
                                title={'Tambah'}
                                onClick={() => {
                                    this.setCount(count + 1);
                                }}
                            />
                            <Button
                                title={'Kurang'}
                                color={'#ff0000'}
                                onClick={() => {
                                    this.setCount(count - 1);
                                }}
                            />
                        </View>
                    </View>
                </SafeAreaView>
            </Fragment>
        );
    }
}

const style = StyleSheet.create({
    controllerViewStyle: {
        alignItems: 'center',
        justifyContent: 'space-around',
        flexDirection: 'row',
    },
});

