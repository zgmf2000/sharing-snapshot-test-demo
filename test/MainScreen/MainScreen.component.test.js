import React from 'react';
import {configure, shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import MainScreen from '../../src/MainScreen/MainScreen.component';

configure({ adapter: new Adapter() });

describe('MainScreen', () => {
  describe('#render', () => {
    it('should render the component according to the snapshot', () => {
      const wrapper = renderer.create(<MainScreen/>);
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

  });
  describe('#counterFunctionallity', () => {
    it('should do addition to count if Tambah Button is pressed',() => {
      const wrapper = shallow(<MainScreen/>);
      const additionButton = wrapper.find({title: 'Tambah'});

      additionButton.props().onClick();
      expect(wrapper.state().count).toEqual(1);
    });

    it('should do addition to count if Tambah Button is pressed',() => {
      const wrapper = shallow(<MainScreen/>);
      const additionButton = wrapper.find({title: 'Kurang'});

      additionButton.props().onClick();
      expect(wrapper.state().count).toEqual(-1);
    });
  });
});
