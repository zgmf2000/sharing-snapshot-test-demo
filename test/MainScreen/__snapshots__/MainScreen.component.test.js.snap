// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen #render should render the component according to the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "marginTop": 25,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "green",
              "fontSize": 30,
              "fontWeight": "normal",
            }
          }
        >
          0
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Tambah
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#ff0000",
                },
              ]
            }
          >
            Kurang
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
